arch/riscv/core/thread.c

```diff
@@ -34,6 +35,8 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 				Z_STACK_PTR_TO_FRAME(struct __esf, stack_ptr)
 				);
 
+	memset(stack_init, 0, sizeof(struct __esf));
```

Comment:
This should be `0xaa` and it's already being done by the kernel in the common code.

Reply:
closing PR then.